akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -14,22 +17,28 @@ import akka.annotation.{ DoNotInherit, InternalApi }
  * If the interceptor does keep mutable state care must be taken to create the instance in a `setup` block
  * so that a new instance is created per spawned actor rather than shared among actor instance.
  *
+ * @param interceptMessageClass Ensures that the interceptor will only receive `O` message types.
+ *                              If the message is not of this class or a subclass thereof it will
+ *                              bypass the interceptor and be continue to the inner behavior untouched.
+ *                              If `null` then all messages bypass the interceptor and only start and
+ *                              signals are intercepted, i.e. aroundReceive for `msg: O` is not invoked.
+ *                              Using `null` for interceptMessageClass is only safe if `O` and `I` are
+ *                              the same.
```

Comment:
Instead of documenting that `null` is supported we can have a separate `BehaviorSignalInterceptor[M]` that should be used to intercept signals only. We use that internally in the `PoisonPillInterceptor`.

Reply:
Added separate `BehaviorSignalInterceptor` in 428548d
Pretty nice.